File: forkjoin-apikit-example/src/main/java/org/forkjoin/apikit/example/api/TestNoResultApiController.java
package org.forkjoin.apikit.example.api;

import org.apache.commons.lang3.ArrayUtils;
import org.forkjoin.apikit.core.Account;
import org.forkjoin.apikit.core.Api;
import org.forkjoin.apikit.core.PageResult;
import org.forkjoin.apikit.core.Result;
import org.forkjoin.apikit.example.form.TestForm;
import org.forkjoin.apikit.example.model.TestObject;
import org.forkjoin.apikit.example.model.TestObjectList;
import org.forkjoin.apikit.example.model.User;
import org.springframework.beans.BeanUtils;
import org.springframework.web.bind.annotation.*;

import javax.validation.Valid;
import java.util.Arrays;
import java.util.List;

/**
 * @author zuoge85 on 15/6/11.
 */
@Api
@RestController
@RequestMapping(value = "v1")
public class TestNoResultApiController {

    @RequestMapping(value = "noResult/testObjectList", method = {RequestMethod.POST})
    @Account(false)
    public TestObjectList<User> testObjectList(@Valid TestForm<User> testForm) throws Exception {
        TestObjectList<User> user = new TestObjectList<>();
        BeanUtils.copyProperties(testForm, user);
        user.setBooleanValueArray(Arrays.asList(ArrayUtils.toObject(testForm.getBooleanValueArray())));
        user.setIntValueArray(Arrays.asList(ArrayUtils.toObject(testForm.getIntValueArray())));
        user.setDoubleValueArray(Arrays.asList(ArrayUtils.toObject(testForm.getDoubleValueArray())));
        user.setFloatValueArray(Arrays.asList(ArrayUtils.toObject(testForm.getFloatValueArray())));
        user.setLongValueArray(Arrays.asList(ArrayUtils.toObject(testForm.getLongValueArray())));

        user.setStringValueArray(Arrays.asList(testForm.getStringValueArray()));
        user.setRegDateArray(Arrays.asList(testForm.getRegDateArray()));
        user.setUsers(Arrays.asList(testForm.getUsers()));

        return user;
    }

    /**
     * 添加
     */
    @RequestMapping(value = "noResult/test", method = {RequestMethod.POST})
    @Account(false)
    public TestObject<User> create(@Valid TestForm<User> testForm) throws Exception {
        TestObject<User> user = new TestObject<>();
        BeanUtils.copyProperties(testForm, user);
        return user;
    }

    @RequestMapping(value = "noResult/test/{id}", method = RequestMethod.GET)
    @Account(false)
    public TestObject<User> get(@PathVariable String id) throws Exception {
        return null;
    }

    @RequestMapping(value = "noResult/test", method = RequestMethod.PUT)
    @Account(false)
    public TestObject<User> update(@Valid TestForm<User> testForm) throws Exception {
        return null;
    }

    @RequestMapping(value = "noResult/test", method = RequestMethod.PATCH)
    @Account(false)
    public TestObject<User> patchUpdate(@Valid TestForm<User> testForm) throws Exception {
        return null;
    }

    @RequestMapping(value = "noResult/test/{id}", method = RequestMethod.DELETE)
    @Account(false)
    public Boolean delete(@PathVariable String id) throws Exception {
        return null;
    }

    @RequestMapping(value = "noResult/tests/{id}", method = RequestMethod.DELETE)
    @Account(false)
    public Integer deletes(@PathVariable List<String> id) throws Exception {
        return null;
    }

    @RequestMapping(value = "noResult/search/{id}/{name}", method = RequestMethod.GET)
    @Account(false)
    public TestObject<User> search(@PathVariable String id, @PathVariable String name) throws Exception {
        return null;
    }

    @RequestMapping(value = "noResult/testString/{name}", method = RequestMethod.GET)
    @Account(false)
    public String testString(@PathVariable String name) throws Exception {
        return name;
    }

    @RequestMapping(value = "noResult/testString1/{name}/{age}", method = RequestMethod.GET)
    @Account(false)
    public String testString1(@PathVariable String name, @PathVariable String age) throws Exception {
        return null;
    }

    @RequestMapping(value = "noResult/page", method = RequestMethod.GET)
    @Account(false)
    public PageResult<TestObject> page() throws Exception {
        return null;
    }
}

File: forkjoin-apikit-example/src/test/js/api/TestNoResultApiController.js

import {AbstractApi} from 'apikit-core'

import {requestGroupImpi} from 'apikit-core'


class TestNoResultApiController extends AbstractApi {

    /**
        * 
     *
        * <div class='http-info'>http 说明：<b>Api Url:</b> <b>noResult/testObjectList</b>
    * <ul>
    * <li><b>Form:</b>TestFormtestObjectList</li>
    * <li><b>Model:</b> TestObjectList</li>
    * </ul>
    * </div>
    * @see TestObjectList
    * @see TestForm

     */
    testObjectList = (testForm) => {
        let _path = null;
        return super._request("test", "testNoResultApiController", "POST", "noResult/testObjectList", _path, testForm);
    }


    /**
        * 添加
     *
        * <div class='http-info'>http 说明：<b>Api Url:</b> <b>noResult/test</b>
    * <ul>
    * <li><b>Form:</b>TestFormcreate</li>
    * <li><b>Model:</b> TestObject</li>
    * </ul>
    * </div>
    * @see TestObject
    * @see TestForm

     */
    create = (testForm) => {
        let _path = null;
        return super._request("test", "testNoResultApiController", "POST", "noResult/test", _path, testForm);
    }


    /**
        * 
     *
        * <div class='http-info'>http 说明：<b>Api Url:</b> <b>noResult/test/{id}</b>
    * <ul>
    * <li><b>PathVariable:</b> string id</li>
    * <li><b>Model:</b> TestObject</li>
    * </ul>
    * </div>
    * @see TestObject
    * @see string

     */
    get = (id) => {
        let _path = {};
        _path["id"] = id;
        return super._request("test", "testNoResultApiController", "GET", "noResult/test/{id}", _path, null);
    }


    /**
        * 
     *
        * <div class='http-info'>http 说明：<b>Api Url:</b> <b>noResult/test</b>
    * <ul>
    * <li><b>Form:</b>TestFormupdate</li>
    * <li><b>Model:</b> TestObject</li>
    * </ul>
    * </div>
    * @see TestObject
    * @see TestForm

     */
    update = (testForm) => {
        let _path = null;
        return super._request("test", "testNoResultApiController", "PUT", "noResult/test", _path, testForm);
    }


    /**
        * 
     *
        * <div class='http-info'>http 说明：<b>Api Url:</b> <b>noResult/test</b>
    * <ul>
    * <li><b>Form:</b>TestFormpatchUpdate</li>
    * <li><b>Model:</b> TestObject</li>
    * </ul>
    * </div>
    * @see TestObject
    * @see TestForm

     */
    patchUpdate = (testForm) => {
        let _path = null;
        return super._request("test", "testNoResultApiController", "PATCH", "noResult/test", _path, testForm);
    }


    /**
        * 
     *
        * <div class='http-info'>http 说明：<b>Api Url:</b> <b>noResult/test/{id}</b>
    * <ul>
    * <li><b>PathVariable:</b> string id</li>
    * <li><b>Model:</b> boolean</li>
    * </ul>
    * </div>
    * @see boolean
    * @see string

     */
    delete = (id) => {
        let _path = {};
        _path["id"] = id;
        return super._request("test", "testNoResultApiController", "DELETE", "noResult/test/{id}", _path, null);
    }


    /**
        * 
     *
        * <div class='http-info'>http 说明：<b>Api Url:</b> <b>noResult/tests/{id}</b>
    * <ul>
    * <li><b>PathVariable:</b> string[] id</li>
    * <li><b>Model:</b> number</li>
    * </ul>
    * </div>
    * @see number
    * @see string[]

     */
    deletes = (id) => {
        let _path = {};
        _path["id"] = id;
        return super._request("test", "testNoResultApiController", "DELETE", "noResult/tests/{id}", _path, null);
    }


    /**
        * 
     *
        * <div class='http-info'>http 说明：<b>Api Url:</b> <b>noResult/search/{id}/{name}</b>
    * <ul>
    * <li><b>PathVariable:</b> string id</li>
    * <li><b>PathVariable:</b> string name</li>
    * <li><b>Model:</b> TestObject</li>
    * </ul>
    * </div>
    * @see TestObject
    * @see string
    * @see string

     */
    search = (id, name) => {
        let _path = {};
        _path["id"] = id;
        _path["name"] = name;
        return super._request("test", "testNoResultApiController", "GET", "noResult/search/{id}/{name}", _path, null);
    }


    /**
        * 
     *
        * <div class='http-info'>http 说明：<b>Api Url:</b> <b>noResult/testString/{name}</b>
    * <ul>
    * <li><b>PathVariable:</b> string name</li>
    * <li><b>Model:</b> string</li>
    * </ul>
    * </div>
    * @see string
    * @see string

     */
    testString = (name) => {
        let _path = {};
        _path["name"] = name;
        return super._request("test", "testNoResultApiController", "GET", "noResult/testString/{name}", _path, null);
    }


    /**
        * 
     *
        * <div class='http-info'>http 说明：<b>Api Url:</b> <b>noResult/testString1/{name}/{age}</b>
    * <ul>
    * <li><b>PathVariable:</b> string name</li>
    * <li><b>PathVariable:</b> string age</li>
    * <li><b>Model:</b> string</li>
    * </ul>
    * </div>
    * @see string
    * @see string
    * @see string

     */
    testString1 = (name, age) => {
        let _path = {};
        _path["name"] = name;
        _path["age"] = age;
        return super._request("test", "testNoResultApiController", "GET", "noResult/testString1/{name}/{age}", _path, null);
    }


    /**
        * 
     *
        * <div class='http-info'>http 说明：<b>Api Url:</b> <b>noResult/page</b>
    * <ul>
    * <li><b>Model:</b> TestObject</li>
    * </ul>
    * </div>
    * @see TestObject

     */
    page = () => {
        let _path = null;
        return super._request("test", "testNoResultApiController", "GET", "noResult/page", _path, null);
    }

}

export { TestNoResultApiController };
const testNoResultApiController = new TestNoResultApiController();
testNoResultApiController._init(requestGroupImpi);
export default testNoResultApiController;

